p2p/connection.go

```diff
@@ -164,7 +170,8 @@ func (c *MConnection) OnStart() error {
 	c.BaseService.OnStart()
 	c.quit = make(chan struct{})
 	c.flushTimer = cmn.NewThrottleTimer("flush", c.config.flushThrottle)
-	c.pingTimer = cmn.NewRepeatTimer("ping", pingTimeout)
+	c.pingTimer = cmn.NewRepeatTimer("ping", c.config.pingTimeout)
+	c.pongTimer = cmn.NewThrottleTimer("pong", c.config.pongTimeout)
```

Comment:
Can you explain why the use of a throttle timer here? I think it may be correct, but want to make sure we have common reasoning.

Reply:
> I think it may be correct

why we might need throttle timer. we send pings every N sec. it's not like we're firing them in bursts. So logic seems pretty straightforward to me: if we didn't receive pong after M sec after sending ping, disconnect (or punish)